raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java

```diff
@@ -205,22 +208,29 @@ public Boolean required()
             // in Raml 0.8 parameters are optional by default except for uri parameters
             if (getVersion(getNode()) == RAML_10 || isUriParameter(getNode()))
             {
-                return NodeSelector.selectType("required", getNode(), true);
+                return selectType("required", getNode(), true);
             }
-            return NodeSelector.selectType("required", getNode(), false);
+            return selectType("required", getNode(), false);
         }
     }
 
     private boolean isUriParameter(Node node)
     {
-        Node ancestor = NodeSelector.selectFrom("../../../../uriParameters", node);
+        Node ancestor = selectFrom("../../../../uriParameters", node);
         return ancestor != null;
     }
 
     public String defaultValue()
     {
-        Object defaultValue = NodeSelector.selectType("default", getNode(), null);
-        return defaultValue != null ? defaultValue.toString() : null;
+        Node result = selectFrom("default", getNode());
+        if (result != null && !(result instanceof NullNode))
+        {
+            if (result instanceof SYObjectNode)
+                return result.toString();
+            if (result instanceof SimpleTypeNode)
+                return ((SimpleTypeNode) result).getValue().toString();
+        }
+        return null;
```

Comment:
What's the value returned if the default value is an array?

On the other hand, the check `!(result instanceof NullNode)` seems redundant, because a string value is only returned for `SYObjectNode` and `SimpleTypeNode`.